src/etc/inc/service-utils.inc

```diff
@@ -700,14 +700,12 @@ function service_control_stop($name, $extras) {
 			sigkillbyname("relayd", "TERM");
 			break;
 		case 'syslogd':
-			if (isvalidpid("{$g['varrun_path']}/syslog.pid")) {
-				sigkillbypid("{$g['varrun_path']}/syslog.pid", "TERM");
-				usleep(100000);
-			}
-			if (isvalidpid("{$g['varrun_path']}/syslog.pid")) {
-				sigkillbypid("{$g['varrun_path']}/syslog.pid", "KILL");
-				usleep(100000);
-			}
+			sigkillbypid("{$g['varrun_path']}/syslog.pid", "TERM");
+			usleep(100000);
+			sigkillbypid("{$g['varrun_path']}/syslog.pid", "KILL");
+			usleep(100000);
```

Comment:
If TERM was successful, and process stopped running no need kill and sleep again? And if for some reason its not running at all, no need to try the TERM+sleep either.? So why make the sleep unconditional? If waiting isn't needed i would prefer to avoid it where possible.

Reply:
It won't kill anything if not running. It does exact same thing like before, without a bunch of duplicate checks. Plus 0.1s delay which of course is a huge deal.